db.go

```diff
@@ -157,6 +158,20 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 		Name: "prometheus_tsdb_retention_cutoffs_failures_total",
 		Help: "Number of times the database failed to cut off block data from disk.",
 	})
+	m.startTime = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "prometheus_tsdb_start_time_seconds",
+		Help: "Oldest timestamp stored in the database.",
+	}, func() float64 {
+		db.mtx.RLock()
+		defer db.mtx.RUnlock()
+		startTime := time.Now().Unix()
```

Comment:
There is no reason to believe that the time stored is going to be the current time. For example, it could `10, 20, 30` or even time in the future. It's best initialise `startTime` as `math.MaxInt64`, and if there is no data in the block, set it to `0`.

Further, the blocks are sorted by time. So doing just `db.blocks[0].MinTime` should suffice. And if `len(db.blocks) == 0`, then we should look at `HeadBlock` mintime.